var/spack/repos/builtin/packages/relion/package.py

```diff
@@ -65,14 +92,20 @@ def cmake_args(self):
             '-DDoublePrec_GPU=%s' % ('+double-gpu' in self.spec),
         ]
 
-        carch = self.spec.variants['cuda_arch'].value
-
         if '+cuda' in self.spec:
-            args += [
-                '-DCUDA=on',
-            ]
-            if carch is not None:
-                args += [
-                    '-DCUDA_ARCH=%s' % (carch),
-                ]
+            # relion+cuda requires selecting cuda_arch
+	    if not carch: 
+		sys.stdout.write('relion+cuda requires selecting cuda_arch')
+		sys.stdout.flush()
+	        sys.exit()
+	    else:
+                args += ['-DCUDA=ON','-DCudaTexture=ON', '-DCUDA_ARCH=%s' % (carch)]
+		sys.stdout.write('cuda arch ' + carch + ' selected')
+		sys.stdout.flush()
```

Comment:
I don't think we need to tell the user what they selected. Also, `print` is much simpler and does the same thing.

Reply:
I tried print but it only prints to the spack build log which does not appear when I then do sys.exit()